src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,238 @@
+        """Lightning Logger function.
+        Initializes and returns the itwinai Logger context for experiment tracking.
+
+        Returns:
+            Logger: The itwinai logger instance.
+        """
+        if not self._initialized:
+            # With the rank_zero_decorator the rank will always be 0
+            self.itwinai_logger.create_logger_context(rank=0)
+            self._initialized = True
+
+        return self.itwinai_logger
+
+    @override
+    @rank_zero_only
+    def finalize(
+        self,
+        status: str
+    ) -> None:
+        """Lightning Logger function.
+        Logs any remaining checkpoints and closes the logger context.
+
+        Args:
+            status (str): Status of the training (e.g., 'completed', 'failed').
+        """
+        if not self._initialized:
+            return
+
+        # Log checkpoints if the last checkpoint was saved but not logged
+        if self._checkpoint_callback:
+            self._scan_and_log_checkpoints(self._checkpoint_callback)
+
+        self.experiment.destroy_logger_context()
+
+    @override
+    @rank_zero_only
+    def log_metrics(
+            self,
+            metrics: Dict[str, float],
+            step: Optional[int] = None
+    ) -> None:
+        """Lightning Logger function.
+        Logs the given metrics and is usually called by the Lightning Trainer.
+
+        Args:
+            metrics (Dict[str, float]): Dictionary of metrics to log.
+            step (Optional[int], optional): Training step associated with the metrics.
+            Defaults to None.
+        """
+        for key, value in metrics.items():
+            self.experiment.log(
+                item=value,
+                identifier=key,
+                kind='metric',
+                step=step
+            )
+
+    @override
+    @rank_zero_only
+    def log_hyperparams(
+            self,
+            params: Union[Dict[str, Any], Namespace]
+    ) -> None:
+        """Lightning Logger function. Logs hyperparameters for the experiment.
+
+        Args:
+            params (Union[Dict[str, Any], Namespace]): Hyperparameters dictionary or object.
+        """
+        if isinstance(params, Namespace):
+            params = vars(params)
+
+        self.experiment.save_hyperparameters(params)
+
+    @override
+    def after_save_checkpoint(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Lightning Logger function. Handles checkpoint saving to the logger after
+        the ModelCheckpoint Callback of the Lightning Trainer is called.
+        The checkpoints are logged as artifacts.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance to manage checkpointing.
+        """
+        if self._log_model == "all" or (
+            self._log_model is True and checkpoint_callback.save_top_k == -1
+        ):
+            self._scan_and_log_checkpoints(checkpoint_callback)
+        elif self._log_model is True:
```

Comment:
I like the use of "is" here, I would not even have thought of that haha.

Reply:
Hahaha

Reply:
I think this is one of the weird habits I have because I find it nicer, but probably not exactly best practice

Reply:
Actually though, because _log_model could also be a string, I think this makes it more clear what we are testing for, does that make sense?

Reply:
I think that `is True` makes the conditional more robust to misuses of user input. Imagine if the user is passing `log_models=42` or `log_models="hello"`... In both cases the control flow could be hijacked.

It seems that this is the same way they used in the MLFlow logger implementation: https://github.com/Lightning-AI/pytorch-lightning/blob/3627c5bfac704d44c0d055a2cdf6f3f9e3f9e8c1/src/lightning/pytorch/loggers/mlflow.py#L328

Reply:
Yeah in this case I specifically liked it because of Lightning's bad choices regarding using just a single variable to control what seems like multiple things. The "is" works as a nice fix for that.